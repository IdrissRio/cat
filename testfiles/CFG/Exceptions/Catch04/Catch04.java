public class Catch04 {
  void test() {
    try {
      try {
        m();
      } catch (TestSubException e) {
        m();
      }
    } catch (TestException e) {
      int a;
    }
    int b;
  }

  void m() throws TestException {}
}

class TestException extends RuntimeException {}
class TestSubException extends TestException {}