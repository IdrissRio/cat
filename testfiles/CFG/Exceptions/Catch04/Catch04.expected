NodeName: Entry[15] Successors: [Exit[16]]
NodeName: Exit[16] Successors: []
NodeName: Entry[7] Successors: [Exit[8]]
NodeName: Exit[8] Successors: []
NodeName: Entry[29] Successors: [MethodAccess[45]]
NodeName: MethodAccess[45] Successors: [UncheckedExceptions[65], UncheckedExceptions[46]]
NodeName: UncheckedExceptions[65] Successors: [UncheckedExceptions[83], Exit[36]]
NodeName: UncheckedExceptions[83] Successors: [VariableDeclarator[89], Exit[36]]
NodeName: VariableDeclarator[89] Successors: [Exit[36]]
NodeName: Exit[36] Successors: []
NodeName: UncheckedExceptions[46] Successors: [ParameterDeclaration[68], Exit[36]]
NodeName: ParameterDeclaration[68] Successors: [VariableDeclarator[79]]
NodeName: VariableDeclarator[79] Successors: [VariableDeclarator[89]]
NodeName: Entry[93] Successors: [Exit[101]]
NodeName: Exit[101] Successors: []
NodeName: Entry[113] Successors: [Exit[114]]
NodeName: Exit[114] Successors: []
NodeName: Entry[105] Successors: [Exit[106]]
NodeName: Exit[106] Successors: []
NodeName: Entry[136] Successors: [Exit[137]]
NodeName: Exit[137] Successors: []
NodeName: Entry[128] Successors: [Exit[129]]
NodeName: Exit[129] Successors: []

NodeName: Exit[16] Predecessor: [Entry[15]]
NodeName: Entry[15] Predecessor: []
NodeName: Exit[8] Predecessor: [Entry[7]]
NodeName: Entry[7] Predecessor: []
NodeName: Exit[36] Predecessor: [UncheckedExceptions[83], UncheckedExceptions[65], UncheckedExceptions[46], VariableDeclarator[89]]
NodeName: UncheckedExceptions[83] Predecessor: [UncheckedExceptions[65]]
NodeName: UncheckedExceptions[65] Predecessor: [MethodAccess[45]]
NodeName: MethodAccess[45] Predecessor: [Entry[29]]
NodeName: Entry[29] Predecessor: []
NodeName: UncheckedExceptions[46] Predecessor: [MethodAccess[45]]
NodeName: VariableDeclarator[89] Predecessor: [UncheckedExceptions[83], VariableDeclarator[79]]
NodeName: VariableDeclarator[79] Predecessor: [ParameterDeclaration[68]]
NodeName: ParameterDeclaration[68] Predecessor: [UncheckedExceptions[46]]
NodeName: Exit[101] Predecessor: [Entry[93]]
NodeName: Entry[93] Predecessor: []
NodeName: Exit[114] Predecessor: [Entry[113]]
NodeName: Entry[113] Predecessor: []
NodeName: Exit[106] Predecessor: [Entry[105]]
NodeName: Entry[105] Predecessor: []
NodeName: Exit[137] Predecessor: [Entry[136]]
NodeName: Entry[136] Predecessor: []
NodeName: Exit[129] Predecessor: [Entry[128]]
NodeName: Entry[128] Predecessor: []
